stable_radical_optimization/stable_radical_opt.py

```diff
@@ -0,0 +1,334 @@
+
+        def windowed_loss(
+                self, target: float,
+                desired_range: Tuple[float, float]) -> float:
+            """ Returns 0 if the molecule is in the middle of the desired range,
+            scaled loss otherwise. """
+
+            span = desired_range[1] - desired_range[0]
+
+            lower_lim = desired_range[0] + span / 6
+            upper_lim = desired_range[1] - span / 6
+
+            if target < lower_lim:
+                return max(1 - 3*(abs(target - lower_lim) / span), 0)
+            elif target > upper_lim:
+                return max(1 - 3*(abs(target - upper_lim) / span), 0)
+            else:
+                return 1
+
+
+    stability_model = tf.keras.models.load_model(stability_model, compile=False)
+    redox_model = tf.keras.models.load_model(redox_model, compile=False)
+    bde_model = tf.keras.models.load_model(bde_model, compile=False)
+
+    # TODO what values should we set for these options?
+    #:param max_atoms: Maximum number of heavy atoms
+    #:param min_atoms: minimum number of heavy atoms
+    #:param atom_additions: potential atom types to consider. Defaults to ('C', 'H', 'O')
+    #:param stereoisomers: whether to consider stereoisomers different molecules
+    #:param sa_score_threshold: If set, don't construct molecules greater than a given sa_score.
+    #:param tryEmbedding: Try to get a 3D embedding of the molecule, and if this fails, remote it.
+    config = MoleculeConfig(max_atoms=15,
+                            min_atoms=4,
+                            tryEmbedding=True,
+                            sa_score_threshold=3.,
+                            stereoisomers=True,
+                            atom_additions=('C', 'N', 'O', 'S'),
+                            )
+
+    #engine = create_engine(f'sqlite:///stable_radical.db',
+    #                       connect_args={'check_same_thread': False},
+    #                       execution_options = {"isolation_level": "AUTOCOMMIT"})
+    dbname = "bde"
+    port = "5432"
+    host = "yuma.hpc.nrel.gov"
+    user = "rlops"
+    # read the password from a file
+    passwd_file = '/projects/rlmolecule/rlops_pass'
+    with open(passwd_file, 'r') as f:
+        passwd = f.read().strip()
+
+    drivername = "postgresql+psycopg2"
+    engine_str = f'{drivername}://{user}:{passwd}@{host}:{port}/{dbname}'
+    engine = create_engine(engine_str, execution_options={"isolation_level": "AUTOCOMMIT"})
+
+    run_id = 'stable_radical_optimization'
+
+    reward_factory = RankedRewardFactory(
+        engine=engine,
+        run_id=run_id,
+        reward_buffer_max_size=250,
+        reward_buffer_min_size=50,
+        ranked_reward_alpha=0.75
+    )
+
+    problem = StableRadOptProblem(
+        engine,
+        config,
+        stability_model,
+        redox_model,
+        bde_model,
+        run_id=run_id,
+        reward_class=reward_factory,
+        # TODO what values should we set for these options?
```

Comment:
Let's start with the defaults from the last run, which would be here
https://github.com/NREL/rlmolecule/blob/augmenting_reward/alphazero/config.py#L25-L27

Reply:
I think I got everything